### ISSUE TITLE
Suggestion: lower the default `array-size-threshold`/make `large_const_arrays` apply to all large consts

### ISSUE THREAD

**Issue by bolshoytoster** [rust-lang/rust-clippy#17074]
I've been [looking into](https://github.com/rust-lang/rust/issues/156911) performance issues with debug builds of [Symphonia](https://github.com/pdeljanov/Symphonia), and it seems it's largely caused by [indexing a large lookup table (const [f32; 512]) in a loop](https://github.com/rust-lang/rust/issues/156911), which causes the entire table to be copied each iteration.

This would have been caught earlier by `clippy::large_const_arrays` if the default threshold was lower.

[I see 162 projects on github using `array-size-threshold`<1kB](https://github.com/search?q=%2Farray-size-threshold+%3D+%5Cd%7B1%2C3%7D%24%2F), indicating that a lot of people prefer to use a smaller threshold.

Lowering the default might cause the `clippy::large_stack_arrays` lint to be triggered more often, so it might be a good idea to have a different threshold for each lint.

Also, would it make sense for `large_const_arrays` to lint all consts larger than `array-size-threshold`? Because currently, this is caught:
```rust
pub const a: [u32; 1_000_000] = [0u32; 1_000_000];
```
but not this:
```rust
pub const a: ([u32; 1_000_000],) = ([0u32; 1_000_000],);
```

**Comment by theirix** [comment]
Taking a look into this issue
